src/components/PostDetails.tsx

```diff
@@ -1,117 +1,134 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { deleteComment } from '../api/posts';
 
-export const PostDetails: React.FC = () => {
-  return (
-    <div className="content" data-cy="PostDetails">
-      <div className="content" data-cy="PostDetails">
-        <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+interface Props {
+  postComments: Comment[],
+  error: string
+  showCommentsLoader: boolean
+  post: Post | undefined,
+  getPostInfo: () => void
+  setError: React.Dispatch<React.SetStateAction<string>>
+}
 
-          <p data-cy="PostBody">
-            eveniet quo quis
-            laborum totam consequatur non dolor
-            ut et est repudiandae
-            est voluptatem vel debitis et magnam
-          </p>
-        </div>
+export const PostDetails: React.FC<Props> = ({
+  postComments,
+  error,
+  showCommentsLoader,
+  post,
+  getPostInfo,
+  setError,
+}) => {
+  const [formIsOpen, setFormIsOpen] = useState(false);
 
-        <div className="block">
-          <Loader />
+  const onRemove = useCallback((id: number) => {
+    deleteComment(id)
+      .then(() => {
+        getPostInfo();
+      })
+      .catch(() => {
+        setError('Cannot delete comment');
+      });
+  }, [postComments]);
 
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
+  const showLoaderComponent = <Loader />;
 
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
+  const showErrorComponent = (
+    <div className="notification is-danger" data-cy="CommentsError">
+      {error}
+    </div>
+  );
+
+  const showNoCommentsComponent = (
+    <div className="title is-4" data-cy="NoCommentsMessage">
+      No comments yet
+    </div>
+  );
 
-          <p className="title is-4">Comments:</p>
+  const CommentsComponent = () => {
+    if (showCommentsLoader) {
+      return <>{showLoaderComponent}</>;
+    }
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+    if (error.length) {
+      return <>{showErrorComponent}</>;
```

Comment:
There is no need to wrap showErrorComponent in a fragment. You can simply return showErrorComponent.